A2rchi/interfaces/chat_app/static/script.js

```diff
@@ -78,10 +86,87 @@ const getChatResponse = async (incomingChatDiv) => {
 
 const copyResponse = (copyBtn) => {
     // Copy the text content of the response to the clipboard
-    const reponseTextElement = copyBtn.parentElement.querySelector("p");
+    const reponseTextElement = copyBtn.parentElement.previousElementSibling.querySelector("p");
     navigator.clipboard.writeText(reponseTextElement.textContent);
 }
 
+const likeResponse = (likeBtn) => {
+    num_responses_since_last_rating = 0;
+
+    const chatContent = likeBtn.parentElement.previousElementSibling.querySelector("p").textContent;
+
+    // fill the image
+    const image = likeBtn.querySelector("img");
+    image.src = "/static/images/thumbs_up_filled.png"
+
+    // make sure other image is not filled
+    const other_image = likeBtn.nextElementSibling.querySelector("img");
+    other_image.src = "/static/images/thumbs_down.png";
+
+    const API_URL = "http://t3desk019.mit.edu:7861/api/like";
+
+     // Send an API request with the chat content and discussion ID
+     fetch(API_URL, {
+        method: "POST", // You may need to adjust the HTTP method
+        headers: {
+            "Content-Type": "application/json",
+            "Access-Control-Allow-Origin": "*",
+        },
+        body: JSON.stringify({ 
+            content: chatContent,
+            discussion_id: discussion_id,
+        }),
+    })
+}
+
+const dislikeResponse = (dislikeBtn) => {
+    num_responses_since_last_rating = 0;
+
+    const chatContent = dislikeBtn.parentElement.previousElementSibling.querySelector("p").textContent;
+
+    // fill the image
+    const image = dislikeBtn.querySelector("img");
+    image.src = "/static/images/thumbs_down_filled.png";
+
+    // make sure other image is not filled
+    const other_image = dislikeBtn.previousElementSibling.querySelector("img");
+    other_image.src = "/static/images/thumbs_up.png";
+
+    const API_URL = "http://t3desk019.mit.edu:7861/api/dislike";
+
+    // Show pop-up form
+    popupForm.style.display = "block";
+
+    //wait for user to submit response
+    submitButton.addEventListener("click", function () {
```

Comment:
I think we may also want to add this event listener on the "X" (i.e. "close popup") button. This way if a user gives a thumbs down -- but no feedback in the popup -- we still record that they disliked the message.